packages/ui/src/Dialog.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, ReactNode, useMemo } from 'react'
+import { Modal, ModalProps, ModalActionProps } from './Modal'
+import { ButtonKind } from './Button'
+
+import styles from './Dialog.module.scss'
+
+export const DIALOG_AFFIRMATION_DEFAULT = 'Are you sure you wish to proceed?'
+
+type DialogActionProps =
+  | {
+      actionOnConfirm: ModalActionProps['onClick']
+      actionChildren: ModalActionProps['children']
+      actionDangerous?: boolean
+      actionDisabled?: ModalActionProps['disabled']
+      actionDisabledTooltip?: ModalActionProps['disabledTooltip']
+    }
+  | {
+      actionDangerous?: never
+      actionDisabled?: never
+      actionDisabledTooltip?: never
+      actionOnConfirm?: never
+      actionChildren?: never
+    }
+
+export type DialogProps = {
+  children?: ReactNode
```

Comment:
What is the use-case for `children`?

Reply:
We support it in `HzConfirmationModal`, but actually, there are no real use-cases. Let me remove that.